### PR TITLE
SNOW-614533: Fix `create_dataframe` that mistakenly converts 0 and False to None when the input data is only a list

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-614533. 

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/udf.py [hunk 1]
```diff
@@ -284,9 +284,9 @@ class UDFRegistration:
             >>> @udf(packages=["numpy"])
             ... def sin_udf(x: float) -> float:
             ...     return np.sin(x)
-            >>> df = session.create_dataframe([0.25 * math.pi, 0.5 * math.pi], schema=["d"])
+            >>> df = session.create_dataframe([0.0, 0.5 * math.pi], schema=["d"])
             >>> df.select(sin_udf("d")).to_df("col1").collect()
-            [Row(COL1=0.7071067811865475), Row(COL1=1.0)]
```
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/udf.py:289]: This one is flaky due to arithmetic precision, so after this bug is fixed, we can use 0 directly to avoid flakiness.

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -1137,10 +1137,12 @@ def convert_row_to_list(
             row: Union[Dict, List, Tuple], names: List[str]
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/session.py:1137]: Nit: please update the type hints.

**Review by sfc-gh-yixie — APPROVED** [review]

File: src/snowflake/snowpark/dataframe_stat_functions.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def approx_quantile(
 
             >>> df = session.create_dataframe([1, 2, 3, 4, 5, 6, 7, 8, 9, 0], schema=["a"])
             >>> df.stat.approx_quantile("a", [0, 0.1, 0.4, 0.6, 1])
-            [0.5, 1.5, 4.5, 6.5, 9.5]
+            [-0.5, 0.5, 3.5, 5.5, 9.5]
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/dataframe_stat_functions.py:45]: Curious why is this changed?

Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/dataframe_stat_functions.py:45]: The original one is not correct https://github.com/snowflakedb/snowpark-python/blob/c838f17c3fb373f7f3cf2192c6a5bf2d74a0dc19/tests/integ/scala/test_dataframe_suite.py#L419
